keylime/da/attest.py

```diff
@@ -0,0 +1,148 @@
+            logger.info(
+                "=> Selected date range for attestation records is: start %d (%s) and end %d (%s)",
+                start_date,
+                datetime.fromtimestamp(start_date).strftime("%Y-%m-%dT%H:%M:%S.%f%z"),
+                end_date,
+                datetime.fromtimestamp(end_date).strftime("%Y-%m-%dT%H:%M:%S.%f%z"),
+            )
+        else:
+            logger.info("=> Will only extract the very last attestation record found on the persistent store")
+
+        if args.agent_uuid.lower() == "all":
+            logger.info("=> Getting a list of agents registered on the persistent store\n")
+            _agent_list = rmc.agent_list_retrieval("auto")
+        else:
+            logger.info("=> Focusing on the agent %s on the persistent store.\n", args.agent_uuid)
+            _agent_list = [args.agent_uuid]
+
+        for _agent_uuid in _agent_list:
+            logger.info("===> Getting all existing registration records for agent %s...\n", _agent_uuid)
+
+            _ak_list = rmc.build_key_list(_agent_uuid)
+
+            logger.info("===> Getting all existing attestation records for agent %s ...\n", _agent_uuid)
+            _attestation_record_list = rmc.record_read(_agent_uuid, start_date, end_date, "verifier")
+
+            logger.info("=====> Verifing the state of agent %s over time...\n", _agent_uuid)
+
+            _p_tpm_ts = None
+            _c_tpm_ts = None
+            _d_tpm_ts = 0
+
+            agentAttestState = None
+            for _record in _attestation_record_list:
+                agent = _record["agent"]
+                json_response = _record["json_response"]
+                ima_policy = _record["ima_policy"]
+
+                logger.info(
+                    "----------- Attesting data (quote and logs from %s, captured by verifier %s (%s:%s)",
+                    _agent_uuid,
+                    agent["verifier_id"],
+                    agent["verifier_ip"],
+                    agent["verifier_port"],
+                )
+
+                if not agentAttestState:
+                    agentAttestState = cloud_verifier_common.get_AgentAttestStates().get_by_agent_id(agent["agent_id"])
+
+                if "tpm_clockinfo" in agent:
+                    if "clock" in agent["tpm_clockinfo"]:
+                        _p_tpm_ts = agent["tpm_clockinfo"]["clock"]
+
+                failure = cloud_verifier_common.process_quote_response(
+                    agent, ima_policy, json_response["results"], agentAttestState
+                )
+
+                if "tpm_clockinfo" in agent:
+                    if "clock" in agent["tpm_clockinfo"]:
+                        _c_tpm_ts = agent["tpm_clockinfo"]["clock"]
+
+                if _p_tpm_ts and _c_tpm_ts:
+                    _d_tpm_ts = _c_tpm_ts - _p_tpm_ts
+
+                quote_failed = False
+                if failure:
+                    if failure.events:
+                        quote_failed = True
```

Comment:
The 4 lines can be expressed as `quote_failed = failure and failure.events` if you want

Reply:
This was how it was originally, `pyright` seemed to disagree, and thus I have changed.

Reply:
Interesting, do you remember the complain? For years that was the correct way, but I should be missing something : ))